lib/semian/sysv_shared_memory.rb

```diff
@@ -0,0 +1,75 @@
+module Semian
+  module SysVSharedMemory #:nodoc:
+    @type_size = {}
+    def self.sizeof(type)
+      size = (@type_size[type.to_sym] ||= (respond_to?(:_sizeof) ? _sizeof(type.to_sym) : 0))
```

Comment:
Why not just use `@type_size[type.to_sym]` and make this easier to read? I assume the only reason you need the `respond_to?` is because you don't have the C implementation just yet?